### ISSUE TITLE
Type qualifiers like `ClassVar` and `Final` should not be allowed in NamedTuples

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#4131]
Both of these field definitions should not be allowed (flagged at runtime and by all other type checkers):

```py
from typing import ClassVar, NamedTuple, Final

class N(NamedTuple):
    a: ClassVar[int]
    b: Final[str] = "foo"
```
